realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -189,6 +189,92 @@ public void onError(Exception e) {
         TestHelper.exitOrThrow(executorService, signalCallbackFinished, signalClosedRealm, backgroundLooper, threadAssertionError);
     }
 
+    // Test if the background Realm is closed when transaction success returned.
+    public void testClosedBeforeAsyncTransactionSuccess() {
+        final AtomicInteger counter = new AtomicInteger(100);
+        final Realm realm = openRealmInstance("testClosedBeforeAsyncTransactionSuccess");
+        final RealmCache.Callback cacheCallback = new RealmCache.Callback() {
+            @Override
+            public void onResult(int count) {
+                assertEquals(1, count);
```

Comment:
use `decrementAndGet` to replace `counter.set(counter.get()-1)` & `if (counter.get() == 0) {`